include/sycl/algorithm/inclusive_scan.hpp

```diff
@@ -80,17 +77,14 @@ OutputIterator inclusive_scan(ExecutionPolicy &sep, InputIterator b,
 
   int i = 1;
   do {
-    auto f = [vectorSize, i, localRange, globalRange, inBuf, outBuf, bop](
+    auto f = [vectorSize, i, ndRange, inBuf, outBuf, bop](
```

Comment:
Please move this lambda out of the loop if it doesn't need to be there.

Reply:
Buffers `inBuf` and `outBuf` are swapped after each iteration. That could probably be written differently but I'd rather not make the change in this PR.

Reply:
In that case, they should be passed as arguments. I agree that it should be a second PR.